assets/admin/editor-wizard/steps/lesson-patterns-step.js

```diff
@@ -15,21 +19,67 @@ import PatternsStep from './patterns-step';
  * @param {Object} props.data Wizard data.
  */
 const LessonPatternsStep = ( { data, ...props } ) => {
+	const { senseiProExtension } = useSelect(
+		( select ) => ( {
+			senseiProExtension: select(
+				EXTENSIONS_STORE
+			).getSenseiProExtension(),
+		} ),
+		[]
+	);
+
 	const replaces = {};
 
 	if ( data.lessonTitle ) {
 		replaces[ 'sensei-content-title' ] = data.lessonTitle;
 	}
 
+	const isSenseiProInstalled =
+		! senseiProExtension || senseiProExtension.is_installed === true;
+
 	return (
-		<PatternsStep
-			title={ __( 'Lesson Type', 'sensei-lms' ) }
-			replaces={ replaces }
-			{ ...props }
-		/>
+		<Fragment>
+			<PatternsStep
+				title={ __( 'Lesson Patterns', 'sensei-lms' ) }
+				replaces={ replaces }
```

Comment:
Wouldn't it be easier to pass here a "showBanner" true/false attribute instead of using the Fills/Slots?

Reply:
Hmmm, originally, I was passing an argument with a function (or undefined) that, if defined, returned a component to inject between the title and the patterns list.

Adding a boolean option to showBanner could be an alternative, but I feel like the Upsell section should be, for now, dedicated to the Lesson Patterns. Given that the Patterns Step is generic (and shared between the Lesson Patterns and the Course Patterns), such an option could create some confusion while reading the code (e.g. as something that is only shown on the lesson patterns, is saved directly in the generic patterns step component). So I'd be a little against that idea.

Considering that, and that passing an argument with a function just to return a component is also, IMHO, a little weird, I decided to go with slot/fill, which, as far as I understand, [has this exact use-case](https://developer.wordpress.org/block-editor/reference-guides/components/slot-fill/).

WDYT? Do you prefer any of these options more strongly? I'm trying to find the more react-y way of doing this here. :P

Reply:
Ok, I forgot that the PatternStep is shared between Lessons and Courses.
I'm ok with your approach then :)